### PR TITLE
Fix libgeotiff dependency of gdal

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gdal/package.py [hunk 1]
```diff
@@ -95,8 +95,8 @@ class Gdal(AutotoolsPackage):
     # Required dependencies
     depends_on('libtiff@3.6.0:')  # 3.9.0+ needed to pass testsuite
     depends_on('libgeotiff@1.2.1:')
-    depends_on('libgeotiff@:1.4', when='@:2.4')
-    depends_on('libgeotiff@1.5:', when='@3:')
+    depends_on('libgeotiff@:1.4', when='@:2.4.0')
+    depends_on('libgeotiff@1.5:', when='@2.4.1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gdal/package.py]: Are you sure? According to https://trac.osgeo.org/gdal/wiki/Release/3.0.0-News#GDALOGR3.0.0-GeneralChanges this change was made in 3.0.0. There's no mention of changing the required libgeotiff version in the 2.4.1 or 2.4.2 release notes.

Comment (iarspider) [reply var/spack/repos/builtin/packages/gdal/package.py]: Sorry - still learning spack version magic. 2.4.1 is compatible with libgeotiff 1.5, but requires 1.2.1+. Is it better now?

File: var/spack/repos/builtin/packages/gdal/package.py [hunk 2]
```diff
@@ -94,9 +94,9 @@ class Gdal(AutotoolsPackage):
 
     # Required dependencies
     depends_on('libtiff@3.6.0:')  # 3.9.0+ needed to pass testsuite
-    depends_on('libgeotiff@1.2.1:')
-    depends_on('libgeotiff@:1.4', when='@:2.4')
-    depends_on('libgeotiff@1.5:', when='@3:')
+    depends_on('libgeotiff@1.2.1:1.4', when='@:2.4.0')
+    depends_on('libgeotiff@1.2.1:1.5', when='@2.4.1')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gdal/package.py]: What about 2.4.2?

File: var/spack/repos/builtin/packages/gdal/package.py [hunk 3]
```diff
@@ -94,9 +94,9 @@ class Gdal(AutotoolsPackage):
 
     # Required dependencies
     depends_on('libtiff@3.6.0:')  # 3.9.0+ needed to pass testsuite
-    depends_on('libgeotiff@1.2.1:')
-    depends_on('libgeotiff@:1.4', when='@:2.4')
-    depends_on('libgeotiff@1.5:', when='@3:')
+    depends_on('libgeotiff@1.2.1:1.4', when='@:2.4.0')
+    depends_on('libgeotiff@1.2.1:1.5', when='@2.4.1:')
+    depends_on('libgeotiff@1.5:', when='@3.0.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gdal/package.py]: Now you're saying that version 3 can only be built with 1.5. I think what you want is:
```python
depends_on('libgeotiff@1.2.1:')
depends_on('libgeotiff@:1.4', when='@:2.4.0')
depends_on('libgeotiff@:1.5', when='@:2')
depends_on('libgeotiff@1.5:', when='@3:')
```

Comment (iarspider) [reply var/spack/repos/builtin/packages/gdal/package.py]: I don't like the hanging `depends_on('libgeotiff@1.2.1:')` - in similar cases I have encountered issues with concretizer ignoring later depends_on clauses. But could be wrong.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gdal/package.py]: If you encounter that problem again, open an issue with instructions for how to reproduce it. The concretizer is far from perfect, but @tgamblin is working on fixing most of the issues.

**Pull request merged**